src/main/java/model/process/analysis/operations/comparisons/TimeBetween.java

```diff
@@ -1,11 +1,168 @@
+				((DataTable) getInput()), ((DataTable) getInput()).getName());
+		builder.addRowsFromTable((DataTable) getInput());
+
+		if (!value.getClass().equals(TimeValue.class)) {
+			builder.addColumn((DataTable) getInput(), new PeriodValue(null,
+					null, null), "Difference date");
+		}
+		if (!value.getClass().equals(DateValue.class)) {
+			builder.addColumn((DataTable) getInput(), new TimeValue(null, null,
+					null), "Difference time");
+		}
+
+		setInput(builder.build());
+
+		DataTable result = getVars((DataTable) getInput());
+
+		return result;
+
+	}
+
+	private DataTable getVars(DataTable table) {
+		
+		if (!value.getClass().equals(TimeValue.class)) {
+			table.getRow(0).setValue(table.getColumn("Difference date"),
+					new PeriodValue(0, 0, 0));
+		}
+		if (!value.getClass().equals(DateValue.class)) {
+			table.getRow(0).setValue(table.getColumn("Difference time"),
+					new TimeValue(0, 0, 0));
+		}
+
+		for (int i = 1; i < table.getRowCount(); i++) {
+			DataRow current = table.getRow(i);
+			DataRow previous = table.getRow(i - 1);
+
+			if (value.getClass().equals(DateTimeValue.class)) {
+				LocalDateTime cur = (LocalDateTime) current.getValue(
+						table.getColumn(dateName)).getValue();
+				curdate = cur.toLocalDate();
+				curtime = cur.toLocalTime();
+
+				LocalDateTime prev = (LocalDateTime) previous.getValue(
+						table.getColumn(dateName)).getValue();
+				predate = prev.toLocalDate();
+				pretime = prev.toLocalTime();
+			}
+			if (value.getClass().equals(DateValue.class)) {
+				curdate = (LocalDate) current.getValue(
+						table.getColumn(dateName)).getValue();
+				predate = (LocalDate) previous.getValue(
+						table.getColumn(dateName)).getValue();
+			}
+			if (value.getClass().equals(TimeValue.class)) {
+				curtime = (LocalTime) current.getValue(
+						table.getColumn(dateName)).getValue();
+				pretime = (LocalTime) previous.getValue(
+						table.getColumn(dateName)).getValue();
+			}
+			
+			table = calculate(table, current);
+			
+		}
+		return table;
+	}
+	
+	private DataTable calculate(DataTable table, DataRow current) {
+		boolean negative = false;
+
+		if (!value.getClass().equals(DateValue.class)) {
+			Duration diffTime = Duration.between(pretime, curtime);
+			int timeConvert = 60;
+			hour = (int) diffTime.toHours();
+			min = ((int) diffTime.toMinutes()) - (hour * timeConvert);
+			sec = ((int) diffTime.getSeconds()) - (min * timeConvert)
+					- (hour * timeConvert * timeConvert);
+			negative = checkNegative();
+			DataValue val = new TimeValue(hour, min, sec);
+			current.setValue(table.getColumn("Difference time"), val);
+		}
+
+		if (!value.getClass().equals(TimeValue.class)) {
+			Period diffDate = Period.between(predate, curdate);
+			if (negative) {
+				diffDate = diffDate.minusDays(1);
+			}
+			DataValue value = new PeriodValue(diffDate.getYears(),
+					diffDate.getMonths(), diffDate.getDays());
+			current.setValue(table.getColumn("Difference date"), value);
+		}
```

Comment:
Als ik je goed begrijp is het probleem dat er geen if statement is welke de DateTimeValue.class afhandelt

if 1 checkt of het geen DateValue is (dus wel TimeValue of DateTimeValue)
if 2 checkt of het geen TimeValue is (dus wel DateValue of DateTimeValue)